Source/ASImageNode.mm

```diff
@@ -533,8 +533,15 @@ - (void)displayDidFinish
   [super displayDidFinish];
 
   __instanceLock__.lock();
-    void (^displayCompletionBlock)(BOOL canceled) = _displayCompletionBlock;
     UIImage *image = _image;
+    void (^displayCompletionBlock)(BOOL canceled) = _displayCompletionBlock;
+    BOOL shouldPerformDisplayCompletionBlock = (image && displayCompletionBlock);
+
+    // Clear the ivar now. The block is retained and will be executed shortly.
+    if (shouldPerformDisplayCompletionBlock) {
+      _displayCompletionBlock = nil;
+    }
+
     BOOL hasDebugLabel = (_debugLabelNode != nil);
```

Comment:
Since many of the calls in the debug label section require the lock anyway (like self.bounds), should we just continue holding the lock until after that?

Reply:
There's a comment below stating that we don't care about locking in the debug section. I don't remember the reasoning behind that (locking violation somehow?), but let's revisit it in another PR.